### ISSUE TITLE
`ripgrep`  depends on "rust" => :build which takes takes 39 minutes

### ISSUE THREAD

**Issue by timotheecour** [Homebrew/homebrew-core#9703]
moved from https://github.com/rust-lang/rust/issues/39576

`brew install rust` takes 39 minutes ; seems by far 1 of the slowest formulas

quoting from https://github.com/rust-lang/rust/issues/39576#issuecomment-277638887:

> The formula for ripgrep contains depends_on "rust" => :build, though, which is surprising and seems to come from here. 

is `rust => :build` needed?

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

```
brew config
start:time=D20170206T165418U272931 hnsecs=636220256582729690 arg:brew config
HOMEBREW_VERSION: 1.1.9-31-g11abfc39a
ORIGIN: https://github.com/Homebrew/brew
HEAD: 11abfc39ab593a525663c9dc7f24a9a93fb895e4
Last commit: 31 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 4d47deeeb2da14f8ba79097a72eaf68efce6867e
Core tap last commit: 3 weeks ago
HOMEBREW_PREFIX: /Users/timothee/homebrew
HOMEBREW_REPOSITORY: /Users/timothee/homebrew
HOMEBREW_CELLAR: /Users/timothee/homebrew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.0 => /Users/timothee/homebrew/bin/git
Perl: /usr/bin/perl
Python: /Users/timothee/homebrew/bin/python => /Users/timothee/homebrew/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_102
macOS: 10.12.2-x86_64
Xcode: 8.2.1
CLT: N/A
X11: 2.7.11 => /opt/X11
MacPorts/Fink: /opt/local/bin/port
```

**Comment by DomT4** [comment]
Your core tap is quite out-of-date there. Nonetheless, for your actual question, `ripgrep` uses `cargo` to build and `cargo` is shipped in the `rust` formula, which for you is building from source because of your non-default prefix & the generated Rust bottles not being relocatable.

**Comment by ilovezfs** [comment]
Why is this confusing? It's written in rust so it needs rust to build.

**Comment by MikeMcQuaid** [comment]
What @DomT4 says.

--- 

@timotheecour Just a heads up that you've been filing a lot of low quality issues which aren't helpful. Please only file issues in future you feel are actionable and clearly demonstrate a bug, thanks.

**Comment by timotheecour** [comment]
@MikeMcQuaid here are more details: From the discussion here https://github.com/rust-lang/rust/issues/39576 which mentioned:

> formula for ripgrep contains depends_on "rust" => :build, though, which is surprising
> I assume :build means "compile" in this context. And compiling Rust is not necessary: You just need the compiler binary and are good to go

I was assuming that meant that `"rust" => :build` could be avoided and written differently so taht brew doesn't try to build `rust` if rust is available in user's PATH; I'm assuming `rust => :build` would always build when there was no bottle (or non-default homebrew install location), even though `rust` may be already available

* for OSX: this would allow skipping the 39m required to install `rust`

* for linuxbrew (yes, i know this is not the place to discuss linuxbrew, but mentioning here as it's related to this thread: removing `rust => :build` actually fixes `ripgrep` (which was broken because `rust` is broken on linuxbrew), I sent out a PR here: https://github.com/Linuxbrew/homebrew-core/pull/1762 that does the fix

**Comment by MikeMcQuaid** [comment]
> even though rust may be already available (as non-brew binary)

This is something we've supported previously we requirements (e.g. if we added `depends_on :rust`) but is something we wish to do less rather than more of. You'll struggle to find (m)any other package managers that allow dependency resolution using arbitrary things in your PATH.

> for OSX: this would allow skipping the 39m required to install rust

We provide a binary package for Rust if you use the default, recommended Homebrew prefix so it doesn't take that long.

> for linuxbrew

To be more explicitly publicly: that Linuxbrew uses a fork of homebrew-core and merges our changes in is in my opinion a big mistake. It's why homebrew-core on Linuxbrew is hit-or-miss quality wise. They should run their own, independent tap and verify changes using CI that's run on Linux. As a result, as you point out, I have literally no interest in discussing Linuxbrew in either here or Homebrew/brew.
